Code/Framework/AzToolsFramework/AzToolsFramework/ViewportSelection/EditorBoxSelect.cpp

```diff
@@ -91,7 +91,8 @@ namespace AzToolsFramework
             debugDisplay.SetLineWidth(s_boxSelectLineWidth);
             debugDisplay.SetColor(s_boxSelectColor);
 
-            AZ::Vector2 viewportSize = AzToolsFramework::GetCameraState(viewportInfo.m_viewportId).m_viewportSize;
+            const AZ::Vector2 viewportSize =
+                AzFramework::Vector2FromScreenSize(AzToolsFramework::GetCameraState(viewportInfo.m_viewportId).m_viewportSize);
```

Comment:
I'm not a fan of the line bloat to go back to a vec2 here.

Reply:
I think if you're adding the GetWidthf accessor then adding the GetAsVector2 seems reasonable.

At this point though I definitely think we're well into the weeds of is it better to increase the api size of the class for niceties or are we just making the class messier to save some characters elsewhere?

Perhaps the best solution is just to do the conversion on a separate line from reading the camera state.

Reply:
Yeah moving to a separate line might just be the cleanest, lowest impact option right now I agree